src/crimson/os/seastore/segment_manager.cc

```diff
@@ -63,18 +63,26 @@ SegmentManager::get_segment_manager(
 LOG_PREFIX(SegmentManager::get_segment_manager);
   return seastar::do_with(
     static_cast<size_t>(0),
-    [&](auto &nr_zones) {
+    [FNAME,
+     dtype,
+     &device](auto &nr_zones) {
```

Comment:
"device" maybe local variable "path" when call make_device, so I think after call "return seastar::do_with", it has chance to get invalid,  better use value copy here and after.

Reply:
I have copied it by value  now, thanks for the feedback.